docs/installation/introduction.rst

```diff
@@ -1,6 +1,104 @@
+picked for optimal on-premises deployments, including pre-configured monitoring
+and alerting, self-healing system configuration, and more.
+
+The installation of a MetalK8s cluster can be broken down into
+the following steps:
+
+#. :doc:`Setup <./setup>` of the environment (with requirements and example
+   OpenStack deployment)
+#. :doc:`Deployment <./bootstrap>` of the :term:`Bootstrap node`, the first
+   machine in the cluster
+#. :doc:`Expansion <./expansion>` of the cluster, orchestrated from the
+   Bootstrap node
+
+.. _Kubernetes: https://kubernetes.io/
+
+.. _installation-intro-architecture:
+
+Choosing a Deployment Architecture
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Before starting the installation, choosing an architecture is recommended, as
+it can impact sizing of the machines and other infrastructure-related details.
+
+.. note:: "Machines" may indicate bare-metal servers or VMs interchangeably
+
+Standard Architecture
+"""""""""""""""""""""
+MetalK8s is designed with on-premise, offline, reliable deployments in mind.
+The first example architecture proposed follows this spirit, focusing on
+reliability rather than compacity:
+
+- One machine dedicated to running Bootstrap services (see
+  :ref:`the Bootstrap role<node-role-bootstrap>` definition below)
+- Three extra machines (or five if installing a really large cluster,
+  100+ nodes) for running the Kubernetes_ control plane (with
+  :ref:`core K8s services<node-role-master>` and the backing
+  :ref:`etcd DB<node-role-etcd>`)
+- One or more machines dedicated to running Infra services (see
+  :ref:`the Infra role<node-role-infra>`)
+- Any number of machines dedicated to running applications, the number and
+  :ref:`sizing<installation-intro-sizing>` depending on the applications (for
+  instance, Zenko_ would recommend using three or more machines)
+
+.. image:: img/standard-arch.png
+   :width: 100%
+
+.. _Zenko: https://zenko.io/
+
+.. _installation-intro-compact-arch:
+
+Compact Architectures
+"""""""""""""""""""""
+While not being focused on having the smallest compute and memory footprints,
+MetalK8s can provide a fully functional single node "cluster". The Bootstrap
+node can be configured to also allow running applications next to all the other
+services required (see :ref:`the section about taints<node-taints>` below).
+
+A single node cluster does not provide any form of resilience to machine or
+site failure, which is why the recommended most compact architecture to use in
+production includes three machines:
+
+- Two machines running control plane services alongside infra and workload
+  applications
+- One machine running Bootstrap services in addition to all the other services
+
+.. image:: img/compact-arch.png
+   :width: 100%
+
+Please note that sizing of such compact clusters needs to account for the
+expected load, and the exact impact of colocating an application with MetalK8s
+services needs to be evaluated by said application's provider.
+
+Variations
```

Comment:
I would think that this architecture is the standard/default one we should recommend. It is also the easiest one to achieve using the tooling and UI we have today. @xaf-scality shouldn't we suggest this one as the default one?